### ISSUE TITLE
Screen readers fail to read some menu items in some context menus of OBS Studio

### ISSUE THREAD

**Issue by k-kolev1985** [obsproject/obs-studio#3363]
### Description
As the title states, screen readers like NVDA, Narrator and JAWS fail to read some menu items in some context menus of OBS Studio. The one noticed is the context menu for a source in the list of sources in the main window of OBS Studio, but I don't know if it is the only one that is problematic. And I just found out, that the problem occurs only when using the high-contrast black theme in Windows, as I do use it.

### Steps to reproduce it
1. Enable high-contrast mode in Windows 10, by doing the following:
1.1. Open Windows Settings (Windows+I).
1.2. Go to Ease of access -> High-contrast.
1.3. Turn on the high-contrast mode, using the respective toggle.
1.4. From the "High-contrast theme" combo-box, select "High-Contrast Black".
2. Launch a screen reader like NVDA, Narrator or JAWS.
3. Launch OBS Studio.
4. Create a scene and add at least one source to it.
5. Invoke the context menu for the added source and use the up/down arrow keys to navigate between its menu items.

### Expected result
The screen readers should properly report all menu items (e.g. its label, its state (inactive/unavailable), if it has a submenu).

### Actual result
The screen readers do not properly report some menu items (e.g. its label, its state (inactive/unavailable), if it has a submenu).

### Test environment
- Operating system: Windows 10 Pro version 2004 (build 19041.450), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".
- OBS Studio version: 25.0.8 (64 bit), portable, in english.
- Processor: Intel Core i5-9400F at 2.90GHz.
- RAM Memory: 16.00GB.
- Graphics: MSI GeForce GTX-1050TI Gaming X 4G, 4096MB dedicated memory, desktop resolution set to 1920x1080.
- Sound Card: Realtek ALC892 at Intel Cannon Point PCH.

**Comment by RytoEX** [comment]
I've tried to check this in OBS Studio 29.1.3 and a recent build from git on Windows 11, and Narrator reads some of the menu items fine (label, state, and if it is collapsed/expanded). The first item in the submenu is "Add", which has a submenu. Narrator reads it as, "Add, menu item, collapsed". It correctly recognizes "Paste (Reference)" as a disabled menu item. In a quick test, it appeared that Narrator correctly reads all menu items, at least the first time I open the context menu.

However, if I opened the context menu more than once, Narrator would *not* read the following items in the source context menu:
* Copy
* Paste (Reference)
* Paste (Duplicate)
* Copy Filters
* Paste Filters
* Rename
* Remove
* Order
* Transform

Curiously, in the Scenes context menu, Rename and Remove suffer the same issue:
https://github.com/obsproject/obs-studio/blob/23bfb625ceb596c60334a282e9cb89efed743fd4/UI/window-basic-main.cpp#L415-L425
https://github.com/obsproject/obs-studio/blob/23bfb625ceb596c60334a282e9cb89efed743fd4/UI/window-basic-main.cpp#L5486-L5487
https://github.com/obsproject/obs-studio/blob/23bfb625ceb596c60334a282e9cb89efed743fd4/UI/window-basic-main.cpp#L5985-L5992

It seems like adding QActions that were defined elsewhere (either via a UI form or by a previous definition) to context menus causes this behavior. Incidentally, if you open a context menu, then close it, then open the Edit menu, the same behavior occurs for items that are shared between the context menu and the Edit menu. If we manually create the QActions on-demand when they are needed, these problems go away.

I have a branch where I've tackled these issues, except for the Transform sub-menu. I don't know if it's PR-ready, so I'll just link it here for now. This seems like it ought to be a bug in Qt, but we'll have to make an example and write it up to find out.
https://github.com/RytoEX/obs-studio/tree/fix-screen-reader-issues

**Comment by RytoEX** [comment]
After some searching, I think these are good QTBUG candidates for the described behavior:
* https://bugreports.qt.io/browse/QTBUG-30818
* https://bugreports.qt.io/browse/QTBUG-79858

**Comment by Fenrirthviti** [comment]
Closing as stale. If this is still an issue please open a new issue with updated reproduction steps on the latest version of OBS.

We have made several updates to Qt since this was reported, and this may already be fixed.
